src/components/views/directory/NetworkDropdown.js

```diff
@@ -158,33 +139,21 @@ export default class NetworkDropdown extends React.Component {
             servers.unshift(MatrixClientPeg.getHomeServerName());
         }
 
-        // if the thirdparty/protocols entries have instance_ids,
-        // we can get the local server listings from here. If not,
-        // the server is too old.
-        let use_protocols = true;
-        for (const proto of Object.keys(this.props.protocols)) {
-            if (!this.props.protocols[proto].instances) continue;
-            for (const instance of this.props.protocols[proto].instances) {
-                if (!instance.instance_id) use_protocols = false;
-            }
-        }
-
         // For our own HS, we can use the instance_ids given in the third party protocols
-        // response to get the server to filter the room list by network for us (if the
-        // server is new enough), although for now we prefer the config if it exists.
-        // For remote HSes, we use the data from the config.
+        // response to get the server to filter the room list by network for us.
+        // We can't get thirdparty protocols for remote server yet though, so for those
+        // we can only show the default room list.
         for (const server of servers) {
-            options.push(this._makeMenuOption(server, null));
-            if (this.props.config.serverConfig && this.props.config.serverConfig[server] && this.props.config.serverConfig[server].networks) {
-                for (const network of this.props.config.serverConfig[server].networks) {
-                    options.push(this._makeMenuOption(server, network));
-                }
-            } else if (server == MatrixClientPeg.getHomeServerName() && use_protocols) {
-                options.push(this._makeMenuOption(server, '_matrix'));
-                for (const proto of Object.keys(this.props.protocols)) {
-                    if (!this.props.protocols[proto].instances) continue;
-                    for (const instance of this.props.protocols[proto].instances) {
-                        options.push(this._makeMenuOptionFromProtocolInstance(server, this.props.protocols[proto], instance));
+            options.push(this._makeMenuOption(server, null, true));
+            if (server == MatrixClientPeg.getHomeServerName()) {
+                options.push(this._makeMenuOption(server, null, false));
+                if (this.props.protocols) {
+                    for (const proto of Object.keys(this.props.protocols)) {
+                        if (!this.props.protocols[proto].instances) continue;
+                        for (const instance of this.props.protocols[proto].instances) {
+                            if (!instance.instance_id) continue;
```

Comment:
In this case we're just iterating through the protocols response to make an option for each, rather than getting one particular protocol / instance